### ISSUE TITLE
Can we respect REAPER's "Sync editor transport to project transport" option in the MIDI Editor?

### ISSUE THREAD

**Issue by ScottChesworth** [jcsteh/osara#929]
As things stand, navigating through chords with OSARA actions always moves the main transport. MIDI Editor -> Options -> "Sync editor transport to project transport" option isn't being respected yet, which can create some awkward UX when navigating tracks/items within the MIDI editor. I'd like to be able to put main transport in a useful spot for auditioning, then zip around tracks and items inside the MIDI Editor, with the main transport staying in the same spot for auditioning the adjustments I'm making. At the moment it's more like make a tweak, cue up audition, make another change, re-cue up audition, make another change, re-cue audition, etc. If the transports could be unsynced, I'd be doing a lot less repetitive button bashing here. Not to add pressure or urgency, but just FYI, the change being proposed already has exuberant support from @timtam and two other buddies of mine who aren't on GitHub, but they're all working with complex MIDI projects on the regular.

If this option can be respected, we'll need a few clarifiers. Reporting the state of the option is easy, it's just a standard toggle. I reckon we'd also need to add a new context to our "OSARA: Report edit/play cursor position and transport state" action. Ideally when the MIDI editor is open and transports aren't linked, it would report MIDI editor transport, reverting to reporting main transport when the editor is closed.

I have no idea how possible or how much work any of this is. Thoughts?

**Comment by ScottChesworth** [comment]
Please remain calm. McLaren only plus 10's here because he can't put hearts on all my Facebook posts anymore.

**Comment by mattymc2** [comment]
They were good times Sent from my iPhoneOn 2 Sep 2023, at 2:04 am, ScottChesworth ***@***.***> wrote:﻿
Please remain calm. McLaren only plus 10's here because he can't put hearts on all my Facebook posts anymore.

**Comment by jcsteh** [comment]
What exactly is the MIDI editor transport? Is this both a separate edit cursor and play cursor in the MIDI editor, keeping mind that the main edit and play cursors are already separate? Once you move the MIDI editor transport, how would you then switch to using the main transport for auditioning as you suggest? Wouldn't that mean closing the MIDI editor anyway, at which point you'd lose your MIDI editor transport?

I don't see any way to access this MIDI editor transport (either edit cursor or play cursor) in the REAPER API.

**Comment by ranetto** [comment]
hello everyone, to fix the problem, I created two actions, one to memorize where I am on which chord or note I am in the midi editor, and one to bring me back to the position I had memorized: in doing so, I can modify, move, listen and return to the position where I was previously, clearly these actions maintain the selection, so I select one or more notes or chords, memorize the selection, and return to it when I need to and I can continue to modify the parameters relating to the group of selected notes. Here are the actions I used for this multi-action:
memory: 
SWS/BR: Save note selection from active item, slot 01  1 di 2
OSARA: Move to next note in chord  2 di 2
Recall:
SWS/BR: Restore note selection to active item, slot 01  1 di 2
Navigate: Move edit cursor to start of selected events  2 di 2

**Comment by ScottChesworth** [comment]
Aha! Neat solution, thanks for sharing!
@timtam and @mattymc2, would this work for you? Seems like it's gonna do what I was hoping for here close Enough.
Seeing as 1) I can't fully picture how this separation of transport could be kept intuitive and 2) Jamie can't see a way to influence MIDI editor transport separately in the API anyway, I'm tempted to rustle up these custom actions, make them available in the Dropbox for anyone who needs the extra functionality and close this as not planned.
Thoughts?

**Comment by jcsteh** [comment]
I don't love that we don't respect a REAPER option, so it 'd be nice to fix this if we could. That said, I still don't understand how this works even for sighted users as per my questions above.

**Comment by mattymc2** [comment]
I’m away at the moment but will give this  a crack when I’m back in the studio. Looks promisingSent from my iPhoneOn 3 Sep 2023, at 8:11 am, James Teh ***@***.***> wrote:﻿
I don't love that we don't respect a REAPER option, so it 'd be nice to fix this if we could. That said, I still don't understand how this works even for sighted users as per my questions above.

**Comment by jcsteh** [comment]
I do see this note in the change log for REAPER 6.74, but its meaning is somewhat unclear and i don't see any API related to the MIDI editor play cursor after searching for quite a while:

`API: add midi editor play cursor preview override API for PlayTime`

**Comment by ScottChesworth** [comment]
PlayTime is an extension for live looping, a sort of Ableton equivalent type thing (inaccessible AFAIK), but yep that does sound like a hopeful lead. Firing off an email to find out more right now.

**Comment by ScottChesworth** [comment]
Change log entry clarified in your inbox @jcsteh (at least, clarified for someone who isn't me) :)

**Comment by jcsteh** [comment]
Yeah, Justin's right, that API isn't at all helpful, even if his answer was. :)

**Comment by ScottChesworth** [comment]
@mattymc2 and @timtam, have you tried the custom actions that @ranetto suggested? Please check https://github.com/jcsteh/osara/issues/929#issuecomment-1703821576 and let us know whether that solution is working well enough.
